### PR TITLE
Expose number of transcoders as Prometheus metric

### PR DESCRIPTION
**Pull request by darkdarkdragon**
**What does this pull request do? Explain your changes. (required)**
Expose number of transcoders currently connected to orchestrator
as Prometheus metric

**Specific updates (required)**

- 
- 
- 

**How did you test each of these updates (required)**
Manually

**Does this pull request close any open issues?**

**Checklist:**

- [ ] README and other documentation updated
- [x] Node runs in OSX and devenv
- [x] All tests in `./test.sh` pass

### CODE REVIEW

File: core/orchestrator.go [hunk 1]
```diff
@@ -803,6 +811,14 @@ func (rtm *RemoteTranscoderManager) completeTranscoders(trans *RemoteTranscoder)
 	sort.Sort(byLoadFactor(rtm.remoteTranscoders))
 }
 
+func (rtm *RemoteTranscoderManager) totalCapacity() int {
```
Comment (j0sh) [inline core/orchestrator.go]: Maybe add a comment that the caller of this function needs to hold the RTmutex

File: core/orchestrator.go [hunk 2]
```diff
@@ -753,13 +753,21 @@ func (rtm *RemoteTranscoderManager) Manage(stream net.Transcoder_RegisterTransco
...
 	rtm.RTmutex.Unlock()
 
 	<-transcoder.eof
 	glog.Infof("Got transcoder=%s eof, removing from live transcoders map", from)
 
 	rtm.RTmutex.Lock()
 	delete(rtm.liveTranscoders, transcoder.stream)
+	if monitor.Enabled {
+		monitor.SetTranscodersNumber(int64(len(rtm.liveTranscoders)))
+		monitor.SetTranscodersCapacity(int64(rtm.totalCapacity()))
```
Comment (j0sh) [inline core/orchestrator.go]: Is this "expensive" ? If so, would it make more sense to store these in temp variables and add to Prometheus outside the lock?

Comment (j0sh) [inline core/orchestrator.go]: Would it also be useful to expose the current load for the transcoder? Or even the load as a percentage of capacity? That seems like good information to track to get a feel for the overall work distribution on the O.

**Review by j0sh — CHANGES_REQUESTED** [review]
Looking good, just asking for some small changes around comments and a few questions.

File: core/orchestrator.go (hunk 1 above)
Comment (darkdarkdragon) [reply core/orchestrator.go]: done

File: core/orchestrator.go (hunk 2 above)
Comment (darkdarkdragon) [reply core/orchestrator.go]: Don't know, but good idea - moved it out.

Comment (darkdarkdragon) [reply core/orchestrator.go]: Good idea. Exposed 'load' metric - just total value of transcoders `load` field. Percentage can be easily calculated in Prometheus itself.

**Comment by darkdarkdragon** [comment]
@j0sh made requested changes

File: core/orchestrator.go [hunk 3]
```diff
@@ -753,14 +753,33 @@ func (rtm *RemoteTranscoderManager) Manage(stream net.Transcoder_RegisterTransco
...
+	var totalLoad, totalCapacity, liveTranscodersNum int
+	if monitor.Enabled {
+		liveTranscodersNum = len(rtm.liveTranscoders)
+		totalLoad, totalCapacity = rtm.totalLoadAndCapacity()
+	}
 	rtm.RTmutex.Unlock()
+	if monitor.Enabled {
+		monitor.SetTranscodersNumber(int64(liveTranscodersNum))
+		monitor.SetTranscodersCapacity(int64(totalCapacity))
+		monitor.SetTranscodersLoad(int64(totalLoad))
```
Comment (j0sh) [inline core/orchestrator.go]: This is starting to feel verbose - would it make sense to have a single function in `monitor` that takes all 3 parameters?

File: monitor/census.go [hunk 4]
```diff
@@ -725,6 +757,24 @@ func TranscodeTry(nonce, seqNo uint64) {
 	}
 }
 
+func SetTranscodersNumber(transcoders int64) {
```
Comment (j0sh) [inline monitor/census.go]: Would it be a little cleaner at the call site to take an `int` type as a parameter, and cast to int64 when calling `Record`? (Same goes for the other functions that were added here)

Comment (darkdarkdragon) [reply monitor/census.go]: done

File: core/orchestrator.go (hunk 3 above)
Comment (darkdarkdragon) [reply core/orchestrator.go]: fixed

**Comment by darkdarkdragon** [comment]
@j0sh done changes

**Review by j0sh — APPROVED** [review]
LGTM after fixing up all the commits into one

**Pull request merged**